GVRf/Framework/framework/src/main/jni/engine/renderer/batch.cpp

```diff
@@ -0,0 +1,197 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "batch.h"
+#include "objects/scene_object.h"
+#include "objects/components/camera.h"
+#include "objects/components/render_data.h"
+#include "objects/light.h"
+
+#define BATCH_SIZE 60
+
+namespace gvr {
+
+Batch::Batch() :
+        draw_count_(0), vertex_count_(0), index_count_(0), vertex_limit_(400),
+        indices_limit_(400), renderdata_(nullptr), index_offset_(0), not_batched_(false) {
+    vertices_.clear();
+    normals_.clear();
+    tex_coords_.clear();
+    indices_.clear();
+    matrices_.clear();
+    matrix_indices_.clear();
+   // mesh = new Mesh(); // should perhaps have a pool of meshes of a certain size
+}
+
+Batch::Batch(int no_vertices, int no_indices) :
+        draw_count_(0), vertex_count_(0), index_count_(0), vertex_limit_(no_vertices),
+        indices_limit_(no_indices), renderdata_(nullptr),
+        index_offset_(0), not_batched_(false) {
+
+    vertices_.clear();
+    normals_.clear();
+    tex_coords_.clear();
+    indices_.clear();
+    matrices_.clear();
+    matrix_indices_.clear();
+    render_data_set_.clear();
+
+    vertices_.reserve(no_vertices);
+    indices_.reserve(no_indices);
+    normals_.reserve(no_vertices);
+    tex_coords_.reserve(no_vertices);
+    matrix_indices_.reserve(no_vertices);
+    glm::mat4 identity;
+   // mesh = new Mesh();
+    glm::vec3 init_vec3(0, 0, 0);
+    glm::vec2 init_vec2(0, 0);
+    for (int i = 0; i < no_vertices; i++) {
+        vertices_.push_back(init_vec3);
+        tex_coords_.push_back(init_vec2);
+        matrix_indices_.push_back(0);
+        normals_.push_back(init_vec3);
+        indices_.push_back(0);
+    }
```

Comment:
What is meant to be stored in vertices_ vector (and other vectors)? What is this used for? What happens if the other ctor is called and we set this data in the mesh.